src/main/java/org/kohsuke/github/GHRelease.java

```diff
@@ -122,41 +122,28 @@ GHRelease wrap(GHRepository owner) {
     }
 
     /**
+     * upload the release using upload URL in {@link upload_url} field.
+     *
      * Because github relies on SNI (http://en.wikipedia.org/wiki/Server_Name_Indication) this method will only work on
      * Java 7 or greater.  Options for fixing this for earlier JVMs can be found here
      * http://stackoverflow.com/questions/12361090/server-name-indication-sni-on-java but involve more complicated
      * handling of the HTTP requests to github's API.
      *
      * @param file file to upload
      * @param contentType content type
-     * @param uploadUrlPrefix URL prefix for uploading; if null {@link #DEFAULT_UPLOAD_URL} will be used
      * @return instance of GHAsset uploaded
-     * @since 1.75
      * @throws IOException
      */
-    public GHAsset uploadAsset(File file, String contentType, String uploadUrlPrefix) throws IOException {
+    public GHAsset uploadAsset(File file, String contentType) throws IOException {
         Requester builder = new Requester(owner.root);
 
-        String url = format("%s%s/releases/%d/assets?name=%s",
-                            defaultIfNull(uploadUrlPrefix, DEFAULT_UPLOAD_URL), owner.getApiTailUrl(""), getId(), file.getName());
+        String url = format(upload_url.replace("{?name}", "?name=%s"), file.getName());
```

Comment:
@kohsuke we can add some more tests here, for example to check if it's not null - if needed?